.github/workflows/main.yml

```diff
@@ -1,16 +1,18 @@
 name: Main Workflow
+
 concurrency:
   group: one-at-time
   cancel-in-progress: false
+
 on:
   push:
     branches:
       - main
   pull_request:
 
 jobs:
-  test:
-    name: Tests
+  checks:
```

Comment:
We optimize the github actions to run in parallel the tests with the checks to speed up waiting time.